src/pipecat/processors/filters/stt_mute_filter.py

```diff
@@ -155,24 +156,32 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
         """Processes incoming frames and manages muting state."""
         await super().process_frame(frame, direction)
 
-        # Handle function call state changes
+        # Handle initial state on StartFrame
+        if isinstance(frame, StartFrame):
+            # Check if we should start muted
+            should_mute = await self._should_mute()
+            if should_mute:
+                await self._handle_mute_state(True)
+
+        # First determine if we need to change mute state
+        should_mute = None
+
```

Comment:
Couldn't this just be like the other ifs below or what am I missing 😅? Like this:
```suggestion
        # First determine if we need to change mute state
        should_mute = None
        
        # Handle initial state on StartFrame
        if isinstance(frame, StartFrame):
            # Check if we should start muted
            should_mute = await self._should_mute()
```